resources/troubleshooting-guides/troubleshooting-push.md

```diff
@@ -0,0 +1,108 @@
+{% hint style="info" %}
+If the current SDK privacy status is `optedout`, the push identifier will not be set.
+{% endhint %}
+
+### setPushIdentifier
+
+{% tabs %}
+
+{% tab title="Android" %}
+
+#### Syntax
+
+```java
+public static void setPushIdentifier(final String pushIdentifier);
+```
+
+#### Example
+
+```java
+// retrieve the token from either GCM or FCM, and pass it to the SDK
+MobileCore.setPushIdentifier(token);
+```
+
+{% endtab %}
+
+{% tab title="objective-c" %}
+
+#### Syntax
+
+```objectivec
++ (void) setPushIdentifier: (nullable NSData*) deviceToken;
+```
+
+#### Example
+
+```objectivec
+// pass the deviceToken that the APNS has assigned to the device
+[ACPCore setPushIdentifier:deviceToken];
+```
+
+{% endtab %}
+
+{% tab title="Swift" %}
+
+#### Syntax
+
+```swift
+Void setPushIdentifier(deviceToken: Data?)
+```
+
+#### Example
+
+```swift
+// pass the deviceToken that the APNS has assigned to the device
+ACPCore.setPushIdentifier(deviceToken)
+```
+
+{% endtab %}
+
+{% tab title="React Native" %}
+
+#### Example
+
+```jsx
+ACPCore.setPushIdentifier("pushIdentifier");
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+### Validate SetPushIdentifier event
+
+Launch your app with the device connected to an [Adobe Griffon session](../../beta/project-griffon). Verify in the list of events that you have an event with type `SetPushIdentifier`. In the details panel on the right, you can verify the value of the push token for this device. The value in `pushIdentifier` is the same value sent to the Adobe servers.
+
+<img src="../../.gitbook/assets/push_token_to_identity.png" />
+
+### Validate Analytics request with Push OptIn
+
+Launch your app with the device connected to an [Adobe Griffon session](../../beta/project-griffon). Verify in the list of events that you have an event with type `AnalyticsForIdentityRequest`. In the details panel on the right, you can see that there is a value sent to Analytics which opts this user in to receive push notifications.
+
+<img src="../../.gitbook/assets/push_analytics_optin.png" />
+
+### Validate the User ID is correct
+
+Launch your app with the device connected to an [Adobe Griffon session](../../beta/project-griffon). Verify in the list of events that you have an event with type `UPDATED_IDENTITY_RESPONSE`. In the details panel on the right, you will want to confirm that two values are correct:
+
+1. The value for `pushidentifier` should match the value sent in step 2 above
+2. The value for `mid` should match the value for `mid` that is sent to Analytics. If you are using a [custom visitor identifier](../../using-mobile-extensions/adobe-analytics/analytics-api-reference#setidentifier), this payload should also contain a `vid` variable with a matching value of that used to identify this user.
+
+<img src="../../.gitbook/assets/push_identities.png" />
+
+<hr />
+
+After completing the above steps, you can be confident that your app is correctly configured to send push messages via the SDK and Adobe.
```

Comment:
Please reword this sentence so that it reads as follows:
"After completing the above steps, your app is correctly configured and is ready to send push messages via the SDK and Adobe."